src/startupscriptgenerator/src/common/buildManifestHelper.go

```diff
@@ -7,49 +7,81 @@ package common
 
 import (
 	"common/consts"
-	"log"
+	"fmt"
+	"os"
 	"path/filepath"
 
 	"github.com/BurntSushi/toml"
 )
 
 type BuildManifest struct {
-	StartupFileName          string
-	ZipAllOutput             string
-	OperationID              string
-	VirtualEnvName           string
-	PackageDir               string
-	CompressedVirtualEnvFile string
-	StartupDllFileName       string
-}
-
-var _buildManifest = BuildManifest{}
-var _readManifest = false
+	// Flag to indicate of the manifest file exists or not
+	Exists bool
 
-func DeserializeBuildManifest(manifestFile string) BuildManifest {
-	var manifest BuildManifest
-	if _, err := toml.DecodeFile(manifestFile, &manifest); err != nil {
-		log.Fatal(err)
-	}
-	return manifest
+	StartupFileName           string
+	ZipAllOutput              string
+	OperationID               string
+	VirtualEnvName            string
+	PackageDir                string
+	CompressedVirtualEnvFile  string
+	StartupDllFileName        string
+	InjectedAppInsights       string
+	CompressedNodeModulesFile string
 }
 
-func GetBuildManifest(appPath string) BuildManifest {
-	if _readManifest {
+var _buildManifest BuildManifest
+var _hasResult = false
+
+func GetBuildManifest(manifestFileDirPathPtr *string, fullAppPath string) BuildManifest {
+	if _hasResult {
 		return _buildManifest
 	}
 
-	tomlFile := filepath.Join(appPath, consts.BuildManifestFileName)
-	if FileExists(tomlFile) {
-		_buildManifest = DeserializeBuildManifest(tomlFile)
-		_readManifest = true
+	manifestFileFullPath := getManifestFile(manifestFileDirPathPtr, fullAppPath)
+	if FileExists(manifestFileFullPath) {
+		fmt.Printf("Found build manifest file at '%s'. Deserializing it...\n", manifestFileFullPath)
+		_buildManifest = deserializeBuildManifest(manifestFileFullPath)
+		_buildManifest.Exists = true
+	} else {
+		fmt.Printf("Cound not find build manifest file at '%s'\n", manifestFileFullPath)
+		_buildManifest.Exists = false
 	}
 
+	_hasResult = true
 	return _buildManifest
 }
 
-func ManifestFileExists(appPath string) bool {
-	tomlFile := filepath.Join(appPath, consts.BuildManifestFileName)
-	fileExists := FileExists(tomlFile)
-	return fileExists
+func getManifestFile(manifestFileDirPathPtr *string, fullAppPath string) string {
+	manifestFileFullPath := ""
+	if *manifestFileDirPathPtr == "" {
+		manifestFileFullPath = filepath.Join(fullAppPath, consts.BuildManifestFileName)
+	} else {
+		providedPath := *manifestFileDirPathPtr
+		absPath, err := filepath.Abs(providedPath)
+		if err != nil || !PathExists(absPath) {
+			fmt.Printf(
+				"Error: Provided manifest file directory path '%s' is not valid or does not exist.\n",
+				providedPath)
+			os.Exit(1)
```

Comment:
Move magic val `1` to a const

Reply:
For now seems like I cannot do it since constants values are currently only 'strings'. Once we fix the constants generating tool to provide ability to specify the type of value, we could fix these.

Reply:
even still, this could be a const in this module (at the top)